estela-api/api/views/auth.py

```diff
@@ -237,81 +243,73 @@ def request(self, request, *args, **kwargs):
+                properties={
+                    "message": openapi.Schema(type=openapi.TYPE_STRING),
+                }
+            ),
+            status.HTTP_400_BAD_REQUEST: openapi.Schema(
+                type=openapi.TYPE_OBJECT,
+                properties={
+                    "message": openapi.Schema(type=openapi.TYPE_STRING),
+                }
+            ),
+        },
+    )
     @action(methods=["GET"], detail=False)
     def validate(self, request, *args, **kwargs):
         token, user_id = self.get_parameters(request)
         user = User.objects.filter(pk=user_id)
         if not user:
-            return redirect(
-                settings.CORS_ORIGIN_WHITELIST[0], {"error": "User does not exist."}
-            )
+            raise UserNotFoundError({"error": "User does not exist."})
         user = user.get()
         if account_reset_token.check_token(user, token):
-            return redirect(
-                settings.CORS_ORIGIN_WHITELIST[0] + "/change_password",
-                {"message": "You can now change your password."},
-            )
+            return Response(data={"message": "Token is valid."}, status=status.HTTP_200_OK)
         else:
-            try:
-                send_change_password_email(user, request)
-            except Exception:
-                raise EmailServiceError(
-                    {
-                        "error": "There was an error sending the verification email. Please try again later."
-                    }
-                )
-            return redirect(
-                settings.CORS_ORIGIN_WHITELIST[0],
-                {"message": "Activation link is invalid. Check your email again."},
-            )
+            return Response(data={"message": "Token is invalid."}, status=status.HTTP_400_BAD_REQUEST)
 
     @swagger_auto_schema(
-        methods=["PATCH"], responses={status.HTTP_200_OK: TokenSerializer()}
+        methods=["PATCH"],
+        manual_parameters=manual_parameters,
+        responses={
+            status.HTTP_200_OK: TokenSerializer(),
+            status.HTTP_401_UNAUTHORIZED: openapi.Schema(
+                type=openapi.TYPE_OBJECT,
+                properties={
+                    "error": openapi.Schema(type=openapi.TYPE_STRING),
+                },
+            ),
+        },
+    )
+    @action(
+        methods=["PATCH"],
+        detail=False,
+        serializer_class=ChangePasswordConfirmSerializer
     )
-    @action(methods=["PATCH"], detail=False)
     def confirm(self, request, *args, **kwargs):
         token, user_id = self.get_parameters(request)
-        old_password = request.data["old_password"]
-        new_password = request.data["new_password"]
-        new_password_repeat = request.data["new_password_repeat"]
         user = User.objects.filter(pk=user_id)
         if not user:
-            raise UserNotFoundError({"error": "User does not exist."})
+            raise UserNotFoundError({
```

Comment:
The `UserNotFoundError` class already contains an error detail attribute. Using `raise UserNotFoundError()` should suffice. If that's the case, please change all the instances where we do this.